### ISSUE TITLE
[AXE-CORE]: The #announcement-root MUST be contained by landmarks

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#7425]
#  [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

While conducting an [audit](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7024) of https://staging.va.gov/disability/view-dependents, I discovered an issue with the announcement component that appears at the bottom of the viewport. There is an axe-core issue, because this `<div id="announcment-root">` is located between landmark areas. This makes it difficult for screen reader and keyboard users to navigate the content, and they may miss this announcement. 

Additionally, on Windows in IE11 (and potentially other Windows browsers) the browser announcements appear in this area and will overlay this announcement, causing it to be overlooked.

### Why it Matters
Navigating a web page is far simpler for screen reader users if the content splits between multiple high-level sections. Content outside of sections is difficult to find, and the content's purpose may be unclear.

Historically, HTML lacked some key semantic markers such as the ability to designate sections of the page as the header, navigation, main content, and footer. Using both HTML5 elements and ARIA landmarks in the same element is considered a best practice, but the future favors using native HTML5 element regions as browser support increases.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to navigate page content by landmarks and not miss relevant announcements.

## Environment

* Operating System: any
* Browser: any
* Screenreading device: any
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/disability/view-dependents/` in browser
2. Note the announcement at the bottom of the viewport (see following screenshot)
3. Start screen reading device
4. Navigate the page by landmarks
5. Verify that the announcement cannot be located
6. Use Windows + IE11
7. Verify that the browser notification overlaps the announcement (see following screenshot)

## Possible Fixes (optional)

If this announcement is important it should be read at the start of page load, and therefore be located in the code either at the top of the `<header>` or near the `<h1>` within `<main>` or other landmark. Visually, it could be positioned as desired, but preferably readily visible. 

## WCAG or Vendor Guidance (optional)

* [All page content must be contained by landmarks, Deque University](https://dequeuniversity.com/rules/axe/3.5/region?application=AxeChrome)
* [ARIA11: Using ARIA landmarks to identify regions of a page](http://www.w3.org/TR/WCAG20-TECHS/ARIA11.html)

## Screenshots or Trace Logs

![Screen Shot 2020-03-30 at 4 51 51 PM](https://user-images.githubusercontent.com/57469/77961375-b8937d80-72a7-11ea-93b5-ad3ddcbe9446.png)
![Screen Shot 2020-03-30 at 4 52 03 PM](https://user-images.githubusercontent.com/57469/77961378-b8937d80-72a7-11ea-9b84-ef4df6e2932d.png)
![image (2)](https://user-images.githubusercontent.com/57469/77961421-c943f380-72a7-11ea-9af2-2b5801b60f42.png)

**Comment by 1Copenut** [comment]
Confirmed on 7/5/2021 these are still valid concerns on VA.gov (prod):

![Screen Shot 2021-07-06 at 9 22 39 AM](https://user-images.githubusercontent.com/934879/124616833-08aea780-de3c-11eb-881c-163e84eb1ef0.png)

**Comment by brianalloyd** [comment]
Thanks @1Copenut for reviewing, incorporating into our burn down now.

**Comment by kelsonic** [comment]
Feel free to update the estimate on this ticket if it wasn't accurate 🙂

**Comment by Rleahy22** [comment]
For validation, the announcement-root in the footer now has a role of `contentinfo`, and the homepage banner now has a role of `region`.

**Comment by brianalloyd** [comment]
@1Copenut the work is done on this, could you review and validate in support of close out?  TIA.

**Comment by 1Copenut** [comment]
@brianalloyd The homepage `banner[role="region"]` is working great in all screen readers.

I'm a bit on the fence about labeling the `section[role="contentinfo"]` for the alert banner at the bottom. I think it would be better to leave the contentinfo role on the `<footer>` tag and add the alert banner in a nested section with an aria-labelledby pattern like what was done on the banner.

LMK next week if you want to chat further about this, or talk with Josh and Angela if I'm not available. Thank you!
